### PR TITLE
Initial take on reducing (cognitive) complexity

### PR DESCRIPTION
**Pull request by j-poecher**
This addresses #412 and will be the first in a row of pull requests

### CODE REVIEW

File: pkg/lib/dynatrace/dashboards.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// DynatraceDashboards is the data structure for /dashboards endpoint
+type DynatraceDashboards struct {
+	Dashboards []struct {
+		ID    string `json:"id"`
+		Name  string `json:"name"`
+		Owner string `json:"owner"`
+	} `json:"dashboards"`
+}
+
+// SearchForDashboardMatching searches for a dashboard that matches either project, service or stage
```
Comment (agrimmer) [inline pkg/lib/dynatrace/dashboards.go:18]: No error was introduced due to the refactoring, but actually, we would like to have an exact match of project, service, and stage. Right now, only either project, service or stage needs to match.

Comment (j-poecher) [reply pkg/lib/dynatrace/dashboards.go]: I have added several tests to verify that only full matches of `project`, `service` and `stage` are allowed. Which is the actual behavior - so I just updated the documentation of the method.

Please have a look at: https://github.com/keptn-contrib/dynatrace-service/pull/415/commits/d5c398a9bc6511d10875a51c3c9d071ade8688f1

Comment (j-poecher) [reply pkg/lib/dynatrace/dashboards.go]: added further tests to also verify that no subsets of `project`, `service` and/or `stage` are matched. Please see: https://github.com/keptn-contrib/dynatrace-service/pull/415/commits/1c0663b1a146f28557350f5783f308056fe9a1e4

Comment (agrimmer) [reply pkg/lib/dynatrace/dashboards.go]: Thank you! 
My assumption was wrong and the tests verified it.

**Review by agrimmer — APPROVED** [review]
LGTM

**Pull request merged**